### PR TITLE
Merge work into main

### LINKED ISSUE #11: Create initial ROI over video

### PR DESCRIPTION
**Pull request by yannickgloster**

### CODE REVIEW

**Review by jackgilbride999 — APPROVED** [review]

File: components/Canvas.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    window.addEventListener("resize", this.handleResize);
+    window.addEventListener("scroll", this.listenToScroll);
+
+    if (this.videoSize.current) {
+      console.log(this.videoSize.current);
+      const dimensions = this.videoSize.current.getBoundingClientRect();
+      this.setState({
+        videoW: dimensions.width,
+        videoH: dimensions.height
```
Comment (LukeHackett12) [inline components/Canvas.js]: videoW and videoH not used, no need to set them

File: components/Canvas.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+const listROIs = [];
+
+class Canvas extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+    this.handlePointerDown = this.handlePointerDown.bind(this);
+    this.handlePointerUp = this.handlePointerUp.bind(this);
+    this.videoSize = React.createRef();
```
Comment (LukeHackett12) [inline components/Canvas.js]: Ref not needed anymore

File: components/Canvas.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      x: 0,
+      y: 0,
+      click: false,
+      overVideo: false,
+      clickX: 0,
+      clickY: 0,
+      windowHeight: 0,
+      windowWidth: 0,
+      width: 0,
+      height: 0,
```
Comment (LukeHackett12) [inline components/Canvas.js]: windowHeight and windowWidth, as well as width and height not needed either

File: components/Canvas.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+class Canvas extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+    this.handlePointerDown = this.handlePointerDown.bind(this);
+    this.handlePointerUp = this.handlePointerUp.bind(this);
+    this.videoSize = React.createRef();
+    this.state = {
+      x: 0,
+      y: 0,
```
Comment (LukeHackett12) [inline components/Canvas.js]: Suggest renaming to mouseX and mouseY

File: components/Canvas.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            (videoElem && videoElem.offsetWidth)) &&
+      (this.state.clickY >
+        (videoElem && videoElem.getBoundingClientRect().top) ||
+        this.state.clickY <
+          (videoElem && videoElem.getBoundingClientRect().top) +
+            (videoElem && videoElem.offsetHeight))
+    ) {
+      return true;
+    } else {
+      return false;
```
Comment (LukeHackett12) [inline components/Canvas.js]: Cleaner to remove else {} and just have 'return false' outside the if statement?

File: components/Canvas.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    }
+  }
+
+  componentDidMount() {
+    this.handleResize();
+    this.listenToScroll();
+
+    this.setState({
+      windowHeight: window.innerHeight,
+      windowWidth: window.innerWidth,
```
Comment (LukeHackett12) [inline components/Canvas.js]: Same as above, no longer needed

File: components/Canvas.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      divPos: ReactDOM.findDOMNode(this).getBoundingClientRect()
+    });
+  };
+
+  listenToScroll = () => {
+    const scrolled = window.pageYOffset;
+
+    this.setState({
+      scrollPos: scrolled
```
Comment (LukeHackett12) [inline components/Canvas.js]: no need for variable just set directly

File: components/Canvas.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    window.removeEventListener("resize", this.handleResize);
+    window.removeEventListener("scroll", this.listenToScroll);
+  }
+
+  handleResize = () => {
+    const videoElem = document.getElementById("react-player");
+    this.setState({
+      videoElem: videoElem,
+      windowHeight: window.innerHeight,
+      windowWidth: window.innerWidth,
```
Comment (LukeHackett12) [inline components/Canvas.js]: Same as above, no longer needed

File: pages/playerLayout.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
+import { Box, Card, Image, Heading, Text, Flex, Button } from "rebass";
```
Comment (LukeHackett12) [inline pages/playerLayout.js]: Maybe we can stash this work then branch off master and work on it from there? Best to keep different aspects seperate IMO.

**Review by LukeHackett12 — CHANGES_REQUESTED** [review]
As a general thing is public/test.txt needed? Also sorry for all the comments haha more just to remember I'll go over it in the morning.

Also just realised definitely need to have the video upload stuff in a different pull request. Should talk about security around it etc in the meeting tomorrow 😃.

**Comment by yannickgloster** [comment]
> 
> 
> As a general thing is public/test.txt needed? Also sorry for all the comments haha more just to remember I'll go over it in the morning.
> 
> Also just realised definitely need to have the video upload stuff in a different pull request. Should talk about security around it etc in the meeting tomorrow 😃.

public/test.txt was just so that the public folder would be created for everyone

File: pages/playerLayout.js (hunk 9 above)
Comment (yannickgloster) [reply pages/playerLayout.js]: Sounds good to me

Comment (yannickgloster) [reply pages/playerLayout.js]: In that case, /pages/api/upload.js should be stashed too

Comment (LukeHackett12) [reply pages/playerLayout.js]: Yep and also some changes to index.js removed

**Review by LukeHackett12 — APPROVED** [review]

**Comment by LukeHackett12** [comment]
Any video uploading stuff was moved to different branch

**Pull request merged**